swift/ql/test/library-tests/dataflow/flowsources/customurlschemes.swift

```diff
@@ -1,26 +1,35 @@
 // --- stubs ---
 class UIApplication {
-    struct OpenURLOptionsKey {}
+    struct OpenURLOptionsKey : Hashable {
+        static func == (lhs: OpenURLOptionsKey, rhs: OpenURLOptionsKey) -> Bool {
+            return true;
+        }
+
+        func hash(into hasher: inout Hasher) {}
+    }
 }
 
 struct URL {}
 
 protocol UIApplicationDelegate {
-    optional func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool
-    optional func application(_ application: UIApplication, handleOpen url: URL) -> Bool
-    optional func application(_ application: UIApplication, open url: URL, sourceApplication: String?, annotation: Any) -> Bool
+    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool
+    func application(_ application: UIApplication, handleOpen url: URL) -> Bool
+    func application(_ application: UIApplication, open url: URL, sourceApplication: String?, annotation: Any) -> Bool
```

Comment:
`optional` is only allowed for `@objc` protocols, which in turn are only allowed with `-enable-objc-interop`, and in any case only by importing `Foundation` that we do not have available in QL tests...

Reply:
I don't think it will matter to what we're testing here.